lgc/include/lgc/patch/PatchLoadScalarizer.h

```diff
@@ -32,30 +32,49 @@
 
 #include "lgc/Builder.h"
 #include "lgc/patch/Patch.h"
+#include "lgc/state/PipelineShaders.h"
+#include "lgc/state/PipelineState.h"
 #include "llvm/IR/InstVisitor.h"
 
 namespace lgc {
 
 // =====================================================================================================================
 // Represents the pass of LLVM patching operations for scalarize load.
-class PatchLoadScalarizer final : public llvm::FunctionPass, public llvm::InstVisitor<PatchLoadScalarizer> {
+class PatchLoadScalarizer final : public llvm::InstVisitor<PatchLoadScalarizer>,
+                                  public llvm::PassInfoMixin<PatchLoadScalarizer> {
 public:
   explicit PatchLoadScalarizer();
 
-  void getAnalysisUsage(llvm::AnalysisUsage &analysisUsage) const override;
-  bool runOnFunction(llvm::Function &function) override;
+  llvm::PreservedAnalyses run(llvm::Function &function, llvm::FunctionAnalysisManager &analysisManager);
 
-  void visitLoadInst(llvm::LoadInst &loadInst);
+  bool runImpl(llvm::Function &function, PipelineState *pipelineState);
 
-  static char ID; // ID of this pass
+  static llvm::StringRef name() { return "Patch LLVM for load scalarizer optimization"; }
 
-private:
-  PatchLoadScalarizer(const PatchLoadScalarizer &) = delete;
-  PatchLoadScalarizer &operator=(const PatchLoadScalarizer &) = delete;
+  void visitLoadInst(llvm::LoadInst &loadInst);
 
+private:
   llvm::SmallVector<llvm::Instruction *, 8> m_instsToErase; // Instructions to erase
   std::unique_ptr<llvm::IRBuilder<>> m_builder;             // The IRBuilder.
   unsigned m_scalarThreshold;                               // The threshold for load scalarizer
 };
 
+// =====================================================================================================================
+// Represents the pass of LLVM patching operations for scalarize load.
+class LegacyPatchLoadScalarizer final : public llvm::FunctionPass {
+public:
+  explicit LegacyPatchLoadScalarizer();
```

Comment:
Is there a reason you are using `explicit` for the default constructor?

Reply:
This is unrelated to this patch, and that code will eventually be removed once the switch to the new pass manager is completed.